modules/react/tabs/lib/Tabs.tsx

```diff
@@ -19,29 +19,121 @@ export interface TabsProps {
   modelHook: useTabsModel,
   subComponents: {
+    /**
+     * `Tabs.List` is a `<div role="tablist">` element. It is a container for {@link TabsItem Tabs.Item}
+     * subcomponents.
+     *
+     * ```tsx
+     * <Tabs.List>{Tabs.Items}</Tabs.List>
+     * ```
+     */
     List: TabsList,
+    /**
+     * `Tabs.Item` is a `<button role="tab">` element. Each `Tabs.Item` is associated with a
+     * `Tabs.Panel` either by a `data-id` attribute or by the position in the list. If a `data-id`
+     * is provided, it must match the `data-id` passed to the corresponding `Tabs.Panel` component.
+     *
+     * ```tsx
+     * <Tabs.Item data-id="first">First Tab</Tabs.Item>
+     * ```
+     */
     Item: TabsItem,
+    /**
+     * `Tabs.Panel` is a `<div role="tabpanel">` element. It is focusable by default. Each
+     * `Tabs.Panel` is controlled by a `Tabs.Item` either by a `data-id` attribute or by the
+     * position in the list. If a `data-id` is provided, it must match the `data-id` passed to the
+     * corresponding `Tabs.Item` component.
+     *
+     * ```tsx
+     * <Tabs.Panel data-id="first">
+     *   {contents of tab panel}
+     * </Tabs.Panel>
+     * ```
+     */
     Panel: TabsPanel,
+    /**
+     * Optional subcomponent of tabs that makes dealing with dynamic tabs easier. `Tabs.Panels` only
+     * works if the model is given an array of items.
+     *
+     * ```tsx
+     * const items = [{ id: '1', name: 'first' }]
+     * <Tabs items={items}>
+     *   {tab list}
+     *   <Tabs.Panels>
+     *     {item => <Tabs.Panel>Contents of {item.name}</Tabs.Panel>}
+     *   </Tabs.Panels>
+     * </Tabs>
+     * ```
+     */
     Panels: TabsPanels,
+    /**
+     * The overflow button of a `Tabs.List`. When a tab list overflows, this overflow button is
+     * rendered. Overflow only works when the model is given an array of `items`.
+     *
+     * ## Usage
+     *
+     * ```tsx
+     * <Tabs items={items}>
+     *   <Tabs.List overflowButton={
+     *     <Tabs.OverflowButton>More</Tabs.OverflowButton>
+     *   }>
+     *     {dynamic tab items}
+     *   </Tabs.List>
+     * </Tabs>
+     * ```
+     */
     OverflowButton: TabsOverflowButton,
+    /**
+     * @deprecated Use `Tabs.Menu.Popper` instead
+     */
     MenuPopper: TabsMenuPopper,
     /**
      * The overflow menu of the Tabs component. If there isn't enough room to render all the tab
      * items, the extra tabs that don't fit will be overflowed into this menu.
-     * @example
+     *
+     * ```tsx
      * <Tabs.Menu.Popper>
      *   <Tabs.Menu.Card>
      *     <Tabs.Menu.List>
-     *       {(item: MyTabItem) => <Tabs.Menu.Item data-id={item.id}>{item.text}</Tabs.Menu.Item>}
+     *       {(item: MyTabItem) => <Tabs.Menu.Item data-id={item.id}>
+     *         {item.text}
+     *       </Tabs.Menu.Item>}
      *     </Tabs.Menu.List>
      *   </Tabs.Menu.Card>
      * </Tabs.Menu.Popper>
+     * ```
      */
-    Menu,
+    Menu: Menu,
```

Comment:
Was this necessary?

Reply:
Yes. There are 2 different `Menu` components. I can only get JSDoc info from symbols.

The following is a `ShorthandPropertyAssignment` and there's only a single symbol associated with it. The doc parser can only pick up on original `Menu` symbol and not the `Tabs.Menu` symbol.

```ts
Menu,
```

The following is a `PropertyAssignment` and there's 2 symbols - the first one is `Tabs.Menu` and the second one is `Menu`. The doc parser is set to extract JSDoc from the `name` of the `PropertyAssignment`. Without this change, the JSDoc for `Tabs.Menu` would actually be the JSDoc for the `Menu` component.

I tried - there's no way to get the JSDoc of the correct symbol without a lot more AST parsing. I use Typescript's type checker to get object properties. This allows us to do something like:

```ts
subComponents: {
  ...otherSubComponents,
  Menu: TabsMenu
}
```

Reply:
FYI, this will bite us sooner or later when developing components and not knowing we need to do this.

One idea would be to pre-process our code with a transform and expand shorthand properties into key/value pairs. I know how to do this in Babel but I'm not sure what that looks like with our tooling.

Reply:
I can add a comment above the line explaining why. Not all shorthand is bad. This is the only case where it makes a difference.